lf/demo/demo_robot.lf

```diff
@@ -2,24 +2,53 @@ target C {
     timeout: 15 sec
 };
 
-reactor Sensors {    
+// This is what we call a driver container. But I think at Luos we will have 2 drivers, one for the button and one for bumper.
+// But perhaps its more lik nodes.
+reactor Sensors {
+    // Those are like pipe beetween reactors?
     output button_out:bool;
     output bumper_out:bool;
     
+    // State are like shared variables?
+
+    // In Luos basic functionalities are represented by models (we call it template sometimes).
+    // For example "State" model is for buttons, and all the values representing the button can be output, input or both. They are associated to already maid "pipe" depending on the format of data (bool, distance, temperature, ...)
+    // To say it differently the "State" type we have already have the "output button_out:bool;" on it. And its a configurable option in this particular type.
+    // This kind of model is related to a container managing all the "pipes" of the model and sharing it with other containers.
```

Comment:
Looks like in Luos, a msg handler is defined in the model, not in the container, is that correct?

I think the idea of having a msg and a msg handler is shared between Luos and LF, although syntactically different (in Luos, msg variable and msg handler are bundled in a model; in LF, they are defined in a reactor, which is similar to a Luos container.)

Reply:
Using models, Yes it's correct. Users steel have the possibility to handle the messages to make some event triggered actions.
But for non pre-defined models users handle msg directly into the container.
The model thing is not officially released yet (it will be in next next release)!